### DISCUSSION TITLE
setPosition does not trigger requestOccurencesHighlight

### DISCUSSION THREAD

**Discussion by siddhant1** [microsoft/monaco-editor#3742] [Q&A] [answered]
The setPosition function moves the position of the cursor but does not highlight Occurences

Playground example:

https://hediet.github.io/monaco-editor/playground.html?source=v0.35.0#XQAAAAI_AQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscrWEItUcqLhiEseIhl2yvSTKsikuAnq3yOVCHCMPueylLD-VoHOlF3w7elBXLBzOq_Ka4_q7xdHjgZ-qzK6kFkzMBm0ZwMw3Gbr1YZ6zmAYxn0cCVF1phVXwEveHgU5jpptNHId3fhAFIUeiyUDKF-NGP89sgNqakT1kv442b9c8lHmcC3iNglRH0u_3lSWFJhSCSbG8qK-4okKyQyw2PYMvEsAMO6eh2mJfje_t9AfBcGlBJ1hJbxT345NtazqmOIt_sk0KxsM__QK84A

**Comment by hediet** [comment]
@alexdima is this a bug or a feature?

@siddhant1 can you try to debug this?

**Comment by alexdima** [comment, marked as answer]
This is intentional. The occurrence highlighter only runs for explicit cursor gestures -- https://github.com/microsoft/vscode/blob/d912e6c92514b3dd524fe6ff1332a06c5f1d5534/src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts#L332-L336

@siddhant1 You can try pretending to be the mouse and use `editor.trigger('mouse', '_moveTo', { position: { column: 13, lineNumber: 1 } })`. [See playground example](https://hediet.github.io/monaco-editor/playground.html?source=v0.35.0#XQAAAAJeAQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscwzX_uRgmSarvBR0nKoyZc2fSR400p_3lSTXZsAIVInmXcEq2IG1mJxe_Cf6sz0PrPJiRFYj3QnqZO50N2V5T4cxLrtGAM4cWH9e1W2bKgMzdEMVlPyB7m2c8U5eGnk2MIPd_XDYhPK-X2qG0lvF6_3uIjtC-nCgbOyRBzoueI6SC_s2PX8vfGwTncLTFwC1Cwf0S-lwWVIVqWb5uajl7hb1rH29qwi_-av7sdAWF7gDOAHufieHPhogln-s1Q8qErgi4D2axNYJJglYfTv8n6JZpZAnyZxVUha1vrHqy1Ov6pac4).
